src/components/Form/formFunctions.ts

```diff
@@ -0,0 +1,181 @@
+    submit = "Please ensure all fields have been entered correctly. Required fields are labelled with an asterisk.",
+    database = "An error has occurred. Please try again later.",
+}
+
+export const numberRegex = /^\d+$/;
+
+export interface CardProps {
+    formErrors: FormErrors;
+    errorSetter: ErrorSetter;
+    fieldSetter: FieldSetter;
+    fields: Fields;
+}
+
+export interface Person {
+    personType: PersonType;
+    age?: number | null;
+    quantity?: number;
+}
+
+export interface FormErrors {
+    fullName: Error;
+    phoneNumber: Error;
+    addressLine1: Error;
+    addressPostcode: Error;
+    adults: Error;
+    numberChildren: Error;
+    nappySize: Error;
+}
+
+export interface Fields {
+    fullName: string;
+    phoneNumber: string;
+    addressLine1: string;
+    addressLine2: string;
+    addressTown: string;
+    addressCounty: string;
+    addressPostcode: string;
+    adults: Person[];
+    numberChildren: number;
+    children: Person[];
+    dietaryRequirements: booleanGroup;
+    feminineProducts: booleanGroup;
+    babyProducts: boolean | null;
+    nappySize: string;
+    petFood: booleanGroup;
+    otherItems: booleanGroup;
+    deliveryInstructions: string;
+    extraInformation: string;
+    attentionFlag: boolean;
+    signpostingCall: boolean;
+}
+
+export const setError = (
+    errorSetter: React.Dispatch<React.SetStateAction<FormErrors>>,
+    errorValues: FormErrors
+): ErrorSetter => {
+    return (errorKey, errorType) => {
+        errorSetter({ ...errorValues, [errorKey]: errorType });
+    };
+};
+
+export const setField = (
+    fieldSetter: React.Dispatch<React.SetStateAction<Fields>>,
+    fieldValues: Fields
+): FieldSetter => {
+    return (fieldKey, newFieldValue) => {
+        fieldSetter({ ...fieldValues, [fieldKey]: newFieldValue });
+    };
+};
+
+const getErrorType = (
+    input: string,
+    required?: boolean,
+    regex?: RegExp,
+    additionalCondition?: (value: string) => boolean
+): Error => {
+    if (required && input === "") {
+        return Error.required;
+    }
+    if (
+        (regex !== undefined && !input.match(regex)) ||
+        (additionalCondition !== undefined && !additionalCondition(input))
+    ) {
+        return Error.invalid;
+    }
+    return Error.none;
+};
+
+export const onChange = (
+    fieldSetter: FieldSetter,
+    errorSetter: ErrorSetter,
+    key: string,
+    required?: boolean,
+    regex?: RegExp,
+    formattingFunction?: (value: string) => Field,
+    additionalCondition?: (value: string) => boolean
+): selectChangeEventHandler => {
+    return (event) => {
+        const input = event.target.value;
+        const errorType = getErrorType(input, required, regex, additionalCondition);
+        errorSetter(key, errorType);
+        if (errorType === Error.none) {
+            const newValue = formattingFunction ? formattingFunction(input) : input;
+            fieldSetter(key, newValue);
+        }
+    };
+};
```

Comment:
I would perhaps rename this function to something along the lines of `getFormValueOnChange` (or something like that) just to make it specific to the use-case of within forms explicit. I found that code like `onChange={onChange(...)}` isn't immediately obvious as to what it's doing